.github/workflows/tests.yml

```diff
@@ -0,0 +1,101 @@
+---
+name: Terraform modules tests
+
+on:
+  schedule:
+    - cron: '0 1 * * 2'
+  workflow_dispatch:
+  pull_request:
+
+
+env:
+  AWS_PROFILE: "infex"
+  AWS_REGION: "eu-central-1"
+
+jobs:
+  integration-test:
+    runs-on: ubuntu-22.04
+    # Other dependencies from hosted runner
+    # AWS CLI
+    # indirectly node used by actions
+
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+
+      - name: Check labels
+        run: |
+          labels=$(jq -r '.pull_request.labels[].name' "$GITHUB_EVENT_PATH")
+          echo "Labels: $labels"
+          if ! echo "$labels" | grep -qE "^testing-allowed$|^testing-ci-not-necessary$"; then
+            echo "ERROR: Pull request must have either 'testing-allowed' or 'testing-ci-not-necessary' label. Then re-run this workflow by pushing a dummy commit."
+            exit 1
+          fi
+
+      - name: Get go.mod details
+        if: "!contains(github.event.pull_request.labels.*.name, 'testing-ci-not-necessary')"
+        uses: Eun/go-mod-details@b719cd324463e2037cf3a0dd1dd6091bdc2730f4 # v1
+        id: go-mod-details
+        with:
+          modfile: ${{ github.workspace }}/test/src/go.mod
+
+      - name: Set up Go environment
+        if: "!contains(github.event.pull_request.labels.*.name, 'testing-ci-not-necessary')"
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5
+        with:
+          go-version: ${{ steps.go-mod-details.outputs.go_version }}
+
+      - name: Set up Terraform (OpenTofu)
+        if: "!contains(github.event.pull_request.labels.*.name, 'testing-ci-not-necessary')"
+        uses: opentofu/setup-opentofu@ae80d4ecaab946d8f5ff18397fbf6d0686c6d46a # v1
+        with:
+          tofu_version: 1.6.2
```

Comment:
if you only need those two specific tools.
I'd recommend getting the version from `.tool-verisons.`

https://github.com/endorama/asdf-parse-tool-versions

Reply:
yes, same works for go.